### ISSUE TITLE
Extracting binarized activations of intermediate layers 

### ISSUE THREAD

**Issue by anarchivetorecall** [larq/larq#652]
Hi, I'm a graduate student studying processing-in-memory-based machine learning accelerator using binarized neural networks.
I am currently using Larq as a software baseline to evaluate my hardware's performance by comparing accuracy, etc.

So the thing is, I'd like to compare the activations of intermediate layers measured by HW to the SW baseline.
Are there any methods to acquire binarized activations of intermediate layers in Larq API? 

```python
import os
import tensorflow as tf
import numpy as np
from tensorflow import keras
from tensorflow.keras import layers
import larq

from keras import backend as K



(train_images, train_labels), (test_images, test_labels) = tf.keras.datasets.mnist.load_data()
train_images = train_images.reshape((60000, 28, 28))
test_images = test_images.reshape((10000, 28, 28))


train_images, test_images = train_images / 127.5 -1, test_images / 127.5 -1
train_images_binary = np.where(train_images > 0, 1, -1)
test_images_binary = np.where(test_images > 0, 1, -1)


model = tf.keras.models.Sequential(
    [
         tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
         larq.layers.QuantDense(
             512,
             kernel_quantizer = "ste_sign",
             kernel_constraint = "weight_clip",
             name = "first_hidden_layer"
             ),
         tf.keras.layers.BatchNormalization(momentum=0.99, scale=False),
         larq.layers.QuantDense(
             512,
             input_quantizer="ste_sign",
             kernel_quantizer="ste_sign",
             kernel_constraint="weight_clip",
             name = "second_hidden_layer"
             ),
         tf.keras.layers.BatchNormalization(momentum=0.99, scale=False),
         larq.layers.QuantDense(
             512,
             input_quantizer="ste_sign",
             kernel_quantizer="ste_sign",
             kernel_constraint="weight_clip",
             name = "third_hidden_layer"
             ),
         tf.keras.layers.BatchNormalization(momentum=0.99, scale=False),
         larq.layers.QuantDense(
             10,
             activation="softmax",
             kernel_quantizer="ste_sign",
             kernel_constraint="weight_clip",
             name = "output_layer"
             )
     ])


model.compile(
    optimizer='adam',
    loss='sparse_categorical_crossentropy',
    metrics=['accuracy'])

larq.models.summary(model)

with larq.context.quantized_scope(True):
    model.save("Larq_Binary_MLP_784_512_512_512_10.h5")
    weights = model.get_weights()
    

model.fit(train_images_binary, train_labels, batch_size=300, epochs=30)

test_loss, test_acc = model.evaluate(test_images_binary, test_labels)

print(f"Test accuracy {test_acc * 100:.2f} %")
```

This is a code that I wrote for implementing MLP (3 hidden layers with 512 neurons) for MNIST classification.

**Comment by Tombana** [comment]
Hi @anarchivetorecall,

I think that if you use explicit quantizer layers (instead of `input_quantizer="ste_sign"`)
```python
y = larq.quantizers.SteSign()(x)
```
and use the Keras functional API instead of the sequential one, then you could add those quantized tensors directly as a model output and view them.

In case you are using Larq Compute Engine, then this PR will be relevant to you: https://github.com/larq/compute-engine/pull/611 (based on this issue: https://github.com/larq/compute-engine/issues/599), made by someone who wanted to access the bitpacked activations directly.

**Comment by anarchivetorecall** [comment]
Thank you so much.
